n_beats/model_keras.py

```diff
@@ -0,0 +1,296 @@
+        time_series_cleaned_forlearning_x = np.zeros((1, backcast_length))
+        time_series_cleaned_forlearning_y = np.zeros((1, forecast_length))
+        j = np.random.randint(backcast_length, time_series_cleaned.shape[0] + 1 - forecast_length)
+        time_series_cleaned_forlearning_x[0, :] = time_series_cleaned[j - backcast_length: j]
+        time_series_cleaned_forlearning_y[0, :] = time_series_cleaned[j:j + forecast_length]
+        x_train = np.vstack((x_train, time_series_cleaned_forlearning_x))
+        y_train = np.vstack((y_train, time_series_cleaned_forlearning_y))
+
+    ts_min = np.min(x_train, axis = 1).reshape((x_train.shape[0], 1))
+    ts_max = np.max(x_train, axis = 1).reshape((x_train.shape[0], 1))
+    x_train = (x_train - ts_min) / (ts_max - ts_min + 1e-8)
+    y_train = (y_train - ts_min) / (ts_max - ts_min + 1e-8)
+    return x_train, y_train
+
+
+def get_xtest_circular(backcast_length, forecast_length):
+    # https://www.mcompetitions.unic.ac.cy/the-dataset/
+
+    # zero-size array to reduction operation minimum which has no identity
+    x_test = np.array([]).reshape(0, backcast_length)
+    y_test = np.array([]).reshape(0, forecast_length)
+    x_tl = []
+    headers = True
+    filename = 'data/val/'
+    with open(filename + 'Daily-test.csv', "r") as file:
+        reader = csv.reader(file, delimiter=',')
+        for line in reader:
+            if not headers:
+                x_tl.append(line)
+            if headers:
+                headers = False
+    x_tl_tl = np.array(x_tl)
+    x_tl_tl = x_tl_tl[:, 1:]
+    for i in range(x_tl_tl.shape[0]):
+        if (len(x_tl_tl[i]) < backcast_length + forecast_length):
+            continue
+        time_series = np.array(x_tl_tl[i])
+        time_series = [float(s) for s in time_series if s != '']
+        time_series_cleaned = np.array(time_series)
+        time_series_cleaned_forlearning_x = np.zeros(
+            (time_series_cleaned.shape[0] + 1 - (backcast_length + forecast_length), backcast_length))
+        time_series_cleaned_forlearning_y = np.zeros(
+            (time_series_cleaned.shape[0] + 1 - (backcast_length + forecast_length), forecast_length))
+        for j in range(backcast_length, time_series_cleaned.shape[0] + 1 - forecast_length):
+            time_series_cleaned_forlearning_x[j - backcast_length, :] = time_series_cleaned[j - backcast_length:j]
+            time_series_cleaned_forlearning_y[j - backcast_length, :] = time_series_cleaned[j: j + forecast_length]
+        x_test = np.vstack((x_test, time_series_cleaned_forlearning_x))
+        y_test = np.vstack((y_test, time_series_cleaned_forlearning_y))
+        
+    ts_min = np.min(x_test, axis = 1).reshape((x_test.shape[0], 1))
+    ts_max = np.max(x_test, axis = 1).reshape((x_test.shape[0], 1))
+    x_test = (x_test - ts_min) / (ts_max - ts_min + 1e-8)
+    y_test = (y_test - ts_min) / (ts_max - ts_min + 1e-8)
+
+    return x_test, y_test
+
+
+def get_accuracy(y_true, y_hat):
```

Comment:
Technically an accuracy is for a classification problem. Here we deal with regression.

Reply:
Updated